### PR TITLE
Use `herculesCI` instead of `flake.herculesCI`

### PR DESCRIPTION
**Pull request by emiflake**
### Summary of changes

Replaces #63. Adding hercules-ci-effects module as dependency that is provided for consumers of liqwid-nix fixes [the issue mentioned in the PR](https://github.com/Liqwid-Labs/liqwid-nix/pull/63#issuecomment-1552079764).

Additionally, we can release 2.9.1 with this.

### Tested on
- [x] [agora](https://github.com/Liqwid-Labs/agora)
- [ ] [liqwid-libs](https://github.com/Liqwid-Labs/liqwid-libs)
- [ ] [oracle-offchain](https://github.com/Liqwid-Labs/oracle-offchain)
- [ ] ...

### CODE REVIEW

File: nix/ci-config.nix [hunk 1]
```diff
@@ -70,13 +70,11 @@ in
     # NOTE(Emily, 22 Mar 2023): Here we provide sensible defaults for working with liqwid-nix. 
     # These can be overwritten by the user, though. In the future, we may want some out-of-the-
     # -box implementation for enabling this somehow.
-    flake = { ... }: {
+    herculesCI = {
       # Added in: 2.7.2.
```
Comment (kylixafonso) [inline nix/ci-config.nix:74]: technically added in 2.9.1 but not super important

Comment (emiflake) [reply nix/ci-config.nix:74]: Well, the functionality was added in 2.7.2 and preserved until now. If we update the version every time we modify it, it is basically useless information. IMO.

Comment (kylixafonso) [reply nix/ci-config.nix:74]: I'd agree if we could guarantee that it works in exactly the same way but down the line if issues arise from this change, given that we version it according to my suggestion, we'll know on what version the breaking change happened.

Comment (emiflake) [reply nix/ci-config.nix:74]: Well, I can add a comment for it.

Comment (kylixafonso) [reply nix/ci-config.nix:74]: That sounds even better, something that explains that this was changed from 2.9.1 but that preserves the original version in which Hercules was added.

**Review by kylixafonso — APPROVED** [review]
LGTM ^^

**Pull request merged**